module/zfs/zcp_iter.c

```diff
@@ -361,11 +360,11 @@ zcp_props_iter(lua_State *state)
 	return (3);
 }
 
-static int zcp_props_list(lua_State *);
-static zcp_list_info_t zcp_props_list_info = {
-	.name = "properties",
-	.func = zcp_props_list,
-	.gc = zcp_props_list_gc,
+static int zcp_user_props_list(lua_State *);
+static zcp_list_info_t zcp_user_props_list_info = {
+	.name = "user_properties",
```

Comment:
are we concerned about changing the name?  Maybe not if `"properties"` isn't part of a release yet?

Reply:
So this code was introduced with the initial channel program commit -> d99a015343425a1c856c900aa8223016400ac2dc

so I assume it has been part of a release. @tonyhutter is this right?
If it is what do you think would be the impact of this change would be?
(e.g. maybe distro packages of zfs are so behind that hadn't had the time to use this feature in LTS releases, etc...)

Reply:
Channel programs arn't in the 0.7.x point releases, so you're free to change the name.

Reply:
Channel programs are now in 0.8 release, so are we concerned about changing this name?

Reply:
@tonyhutter @behlendorf  I'm ok reverting the renaming, just wanted to get your thoughts before doing so.

Reply:
Ended up leaving `user_properties` as is and making `properties` an alias to it. Added a comment about this decision above the declaration of `zcp_props_list_info`, and a mention of the alias in the man page. I've also updated their respective tests, to test both APIs. Let me know what you think.

Reply:
Including a `properties` alias for `user_properties` sounds like a good solution to me.